dvc/lock.py

```diff
@@ -26,10 +27,56 @@ class LockError(DvcException):
     """Thrown when unable to acquire the lock for dvc repo."""
 
 
+class Lock(object):
+    """Class for dvc repo lock.
+
+    Uses zc.lockfile as backend.
+    """
+
+    def __init__(self, lockfile, friendly=False, **kwargs):
+        self._friendly = friendly
+        self.lockfile = lockfile
+        self._lock = None
+
+    @property
+    def files(self):
+        return [self.lockfile]
+
+    def _do_lock(self):
+        try:
+            with Tqdm(
+                bar_format="{desc}",
+                disable=not self._friendly,
```

Comment:
Do we ever need this to be unfriendly?

Reply:
@Suor Yes, in backgroud stuff like analytics/updater sometimes.

Reply:
So people might get stuck daemons collected over time?

Reply:
Locks are the same though. So if we have updater lock stuck then we'll have repo lock stuck and a user will be forced to fix it, which will fix it for daemons too.